pwm-fan.sh

```diff
@@ -174,28 +181,36 @@ fan_run_thermal () {
 		if [[ ${#TEMPS[@]} -gt $TEMPS_SIZE ]]; then
 			TEMPS=(${TEMPS[@]:1})
 		fi
-		if [[ ${TEMPS[-1]} -le ${THERMAL_ABS_THRESH[0]} ]]; then
-			echo ${DC_ABS_THRESH[0]} 2> /dev/null > $CHANNEL_FOLDER'duty_cycle'
-		elif [[ ${TEMPS[-1]} -ge ${THERMAL_ABS_THRESH[-1]} ]]; then
-			echo ${DC_ABS_THRESH[-1]} 2> /dev/null > $CHANNEL_FOLDER'duty_cycle'
-		elif [[ ${#TEMPS[@]} -gt 1 ]]; then
-			TEMPS_SUM=0
-			for TEMP in ${TEMPS[@]}; do
-				let TEMPS_SUM+=$TEMP
-			done
-			# moving mid-point 
-			MEAN_TEMP=$((TEMPS_SUM/${#TEMPS[@]}))
-			DEV_MEAN_CRITICAL=$((MEAN_TEMP-100))
-			X0=${DEV_MEAN_CRITICAL#-}
-			# args: x, x0, L, a, b (k=a/b)
-			MODEL=$(function_logistic ${TEMPS[-1]} $X0 ${DC_ABS_THRESH[-1]} 1 10)
-			if [[ $MODEL -lt ${DC_ABS_THRESH[0]} ]]; then
-				echo ${DC_ABS_THRESH[0]} 2> /dev/null > $CHANNEL_FOLDER'duty_cycle'
-			elif [[ $MODEL -gt ${DC_ABS_THRESH[-1]} ]]; then
-				echo ${DC_ABS_THRESH[-1]} 2> /dev/null > $CHANNEL_FOLDER'duty_cycle'
-			else
-				echo $MODEL 2> /dev/null > $CHANNEL_FOLDER'duty_cycle'
-			fi
+		if [[ ${TEMPS[-1]} -le ${THERMAL_ABS_THRESH_OFF} ]]; then
+		    echo "0" 2> /dev/null > $CHANNEL_FOLDER'duty_cycle'
+		    ON_OFF_STATE=0
+		elif [[ ${TEMPS[-1]} -ge ${THERMAL_ABS_THRESH_ON} ]]; then
+		    ON_OFF_STATE=1
+		fi
```

Comment:
Hello @cgomesu,
I played a little with the `-l` option on my setup, but I was not very satisfied with the results.
Using the NanoPi aluminum heat-sink makes for a fairly large thermal inertia, it takes several minutes for the temperature to reach steady state with my setup.

I tried the following parameters: `-l 120 -d 0 -t 49 -D 100 -T 85`
Because the fan cannot start with 1% duty cycle, we already have some sort of hysteresis, that would make the fan kick in around 52~53°C. Never the less, with those settings, I found that the fan would switch state pretty much every 2 minutes.

It could work better by setting `-l` to an even longer time, but I don't find it to be a very good solution, because this would leave enough time for the CPU to get very hot under high load, until the next update cycle activates the fan.

About your alternative proposal, if I understand correctly, this gives us the regular update rate when the current CPU temp. > `-t` threshold  and a slower (configurable) update rate when the current CPU temp. <= `-t` threshold, right ?
I imagine this solution would have the same drawback of delaying the fan activation when the CPU load becomes high.

I was thinking implementing a single parameters creating the hysteresis around `THERMAL_ABS_THRESH_LOW`, something like:
`-u 5 -t 45 -T 85` that would make the FAN-OFF threshold at 45-5=40°C and the FAN-ON threshold at 45+5=50°C
But I opted for the 2 separate arguments, because I found it more straight forward to implement and more flexible.

I perfectly understand that you want to keep the script as simple as possible and avoid bloating it with many options. Please let me know if you find other alternatives or if you want me to test a little more with the `-l` option.

Reply:
your justification sounds reasonable.  I'll go ahead and mark this part of the review as resolved, meaning we will go forward with your original idea to implement two additional arguments.  

now you just need to address the remaining points in my other reviews.  you can write a separate commit for each and then push to your branch and they will show up here.